### PR TITLE
12691: Improve the PR checklist.

### PR DESCRIPTION
**Pull request by iefremov**
Resolves https://github.com/brave/brave-browser/issues/12691

The idea is to clean up the checklist and leave only relevant
and important entries. Also populate the reviewers section with
frequently overlooked issues.

Resolves 

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [x] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [x] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Review by bbondy — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -3,48 +3,35 @@ Resolves
 
 ## Submitter Checklist:
 
-- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
+- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
 - [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
-- [ ] Added/updated tests for this change (for new code or code which already has tests).
-- Verified that these changes build without errors on
-  - [ ] Android
```
Comment (mihaiplesa) [inline .github/PULL_REQUEST_TEMPLATE.md:10]: might still be useful to have this when mixing local test runs + CI

**Review by mihaiplesa — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -3,48 +3,35 @@ Resolves
...
-  - [ ] Windows
-- Verified that these changes pass automated tests (unit, browser, security tests) on
-  - [ ] iOS
-  - [ ] Linux
-  - [ ] macOS
-  - [ ] Windows
-- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
+- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
+- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`
 - [ ] Ran `git rebase master` (if needed).
```
Comment (antonok-edm) [inline .github/PULL_REQUEST_TEMPLATE.md:10]: Can we do something similar to the first checkbox - describing the state of the PR rather than an "if needed" action? I'm thinking something along the lines of:
> - [ ] The PR is rebased/up to date with the most recent master branch

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -3,48 +3,35 @@ Resolves
...
-- [ ] Ran `git rebase -i` to squash commits (if needed).
-- [ ] Tagged reviewers and labelled the pull request as needed.
 - [ ] Requested a security/privacy review as needed.
-- [ ] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
-- [ ] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
-- [ ] Public documentation has been updated as necessary. For instance:
+- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`) to the associated issue
```
Comment (antonok-edm) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: Would be nice to sort all of the "as needed" tasks after the required tasks.

Mentioning the `OS/...` labels here would be good too.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -3,48 +3,35 @@ Resolves
...
-  - [ ] macOS
-  - [ ] Windows
-- Verified that these changes pass automated tests (unit, browser, security tests) on
-  - [ ] iOS
-  - [ ] Linux
-  - [ ] macOS
-  - [ ] Windows
-- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
+- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
+- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`
```
Comment (antonok-edm) [inline .github/PULL_REQUEST_TEMPLATE.md:9]: Perhaps also `npm run tslint`?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -3,48 +3,35 @@ Resolves
...
+
+- [ ] The associated issue milestone is set to the smallest version that the
+  changes has landed on.
+- [ ] All relevant documentation has been updated, for instance:
   - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
   - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
   - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
   - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
   - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
   - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
```
Comment (diracdeltas) [inline .github/PULL_REQUEST_TEMPLATE.md:32]: not part of this PR, but this should be replaced with https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave

**Review by diracdeltas — APPROVED** [review]
btw this also needs to be in https://github.com/brave/brave-browser/blob/master/.github/PULL_REQUEST_TEMPLATE.md

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (fmarier) [inline .github/PULL_REQUEST_TEMPLATE.md:9]: One problem I commonly have is that I almost never have the browser tests working locally and it's never because of my changes. Maybe we should split that one into a separate item (or just leave it for CI)?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -3,48 +3,35 @@ Resolves
...
+- [ ] Major classes, functions and non-trivial code blocks are well-commented
+- [ ] Changes in component dependencies are properly reflected in `gn`
+- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
+- [ ] Test plan is specified in PR before merging
+
+## After-merge Checklist:
+
+- [ ] The associated issue milestone is set to the smallest version that the
+  changes has landed on.
+- [ ] All relevant documentation has been updated, for instance:
```
Comment (fmarier) [inline .github/PULL_REQUEST_TEMPLATE.md:26]: :100: This is never appropriate to do until after the merge :)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -3,48 +3,35 @@ Resolves
...
   - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
   - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
+  - [ ] https://github.com/brave/brave-browser/wiki/P3A
 
 ## Test Plan:
 
-
-## Reviewer Checklist:
-
-- [ ] New files have MPL-2.0 license header.
```
Comment (fmarier) [inline .github/PULL_REQUEST_TEMPLATE.md:41]: This is a frequently forgotten one. Might be worth keeping until we have it automated in the linter.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (iefremov) [reply .github/PULL_REQUEST_TEMPLATE.md:10]: I believe this state is nearly impossible - master is frequently updated, so in a minute after rebase you are not up-to-date anymore. So the current sentence just reminds that it's useful to rebase once in a while.

And if your RP is seriously outdated, CI wouldn't merge it anyway

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (iefremov) [reply .github/PULL_REQUEST_TEMPLATE.md:41]: linter checks it, at least for C++ files

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (iefremov) [reply .github/PULL_REQUEST_TEMPLATE.md:12]: hmm don't we attach `OS/...` only to issues?..

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (iefremov) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: same here! But I reckon it is still useful to run tests locally just in case you catch an error and save some CI cycles

Comment (iefremov) [reply .github/PULL_REQUEST_TEMPLATE.md]: added

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (iefremov) [reply .github/PULL_REQUEST_TEMPLATE.md]: Replaced

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (iefremov) [reply .github/PULL_REQUEST_TEMPLATE.md]: reordered

Comment (antonok-edm) [reply .github/PULL_REQUEST_TEMPLATE.md]: right - unless I misunderstand somehow, this bullet references "the associated issue"

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (fmarier) [reply .github/PULL_REQUEST_TEMPLATE.md:41]: That explains why I haven't seen it missing from C++ files lately, though I have seen it missing from other file types.

Comment (petemill) [reply .github/PULL_REQUEST_TEMPLATE.md:41]: We can have ts files checked with tslint https://github.com/brave/brave-browser/issues/12724

Comment (iefremov) [reply .github/PULL_REQUEST_TEMPLATE.md:41]: ok, let's keep it until we can lint more file types

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (iefremov) [reply .github/PULL_REQUEST_TEMPLATE.md]: ah, right, added.
